src/core/event_engine/mod.rs

```diff
@@ -0,0 +1,97 @@
+//! Event Engine module
+
+use std::sync::RwLock;
```

Comment:
You have to allign that to `no_std` requirements. I can help you with that if you need.

Reply:
Made changes, and it wasn't the only place where proper `no_std`-compatible imports used. Made changes to all places.